File: project_euler/Problem100/Problem100.cpp
#include <iostream>
using namespace std;

class Problem100 {
public:
    void solution() {
        long long target = 1000000000000L;
        long long index = 1L;
        long long result = 1L;
        while (index < target) {
            long long auxiliary = (index << 1) + result * 3 - 2;
            index = index * 3 + (result << 2) - 3;
            result = auxiliary;
        }
        cout << result << endl;
    }
};

File: project_euler/Problem100/Problem100.py
class Problem100:
    def solution(self) -> None:
        target, index, result = 1000000000000, 1, 1
        while index < target:
            result, index = (index << 1) + result * 3 - 2, index * 3 + (result << 2) - 3
        print(result)
